fix(challenges): Fixed import/export issues in challenges

theoutlander opened the pull request:
#### Pre-Submission Checklist
- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change
- [x] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:
- [x] Tested changes locally.
- [x] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #16235 

#### Description
When the import/export challenges were executed, they were looking for a
require method, export method, and specific class structures when
importing modules. I've introduced the ability to create mock
classes/functions. This is tested locally and everything passes.

**Fixes the following challenges:**
[Understand the Differences Between import and require](https://beta.freecodecamp.org/en/challenges/es6/understand-the-differences-between-import-and-require)
[Use export to Reuse a Code Block](https://beta.freecodecamp.org/en/challenges/es6/use-export-to-reuse-a-code-block)
[Use * to Import Everything from a File](https://beta.freecodecamp.org/en/challenges/es6/use--to-import-everything-from-a-file)
[Create an Export Fallback with export default](https://beta.freecodecamp.org/en/challenges/es6/create-an-export-fallback-with-export-default)
[Import a Default Export](https://beta.freecodecamp.org/en/challenges/es6/import-a-default-export)

CC: @Ethan-Arrowood, @BerkeleyTrue, @QuincyLarson

BerkeleyTrue reviewed (changes requested):
@theoutlander These seem to be specific to the challenges that they address. Instead of hard coding them, these should be written into the actual challenge using the head/tail properties.

theoutlander committed:
fix(challenges): Fixed import export related challenges

Closes #16235

theoutlander commented:
@BerkeleyTrue Thanks for the feedback. I've updated my pull request with the appropriate changes in head and removed my previous changes. 

I noticed that my editor keeps auto-formatting the file. I tried to remove the whitespace auto-formatting three times. However, it seems like Webstorm squeezed that in before I committed. On the bright side, it makes the file consistent.
